### DISCUSSION TITLE
Issues when running create-venv.sh with python version higher than 3.11

### DISCUSSION THREAD

**Discussion by eum602** [NillionNetwork/discussions#32] [Bugs]
When running the step [Open another terminal to create and activate a python virtual environment for Nada program development](https://docs.nillion.com/js-quickstart#5-open-another-terminal-to-create-and-activate-a-python-virtual-environment-for-nada-program-development) the following errors are found:

1.
![image](https://github.com/NillionNetwork/discussions/assets/11699587/a2c1209b-9620-4e69-b5e4-fc19f0c1c113)

Running with **bash** solves the issue:
2. The script `create-venv.sh` assumes python is 3.11 version, but according to the documentation python3.11 and higher is supported. I have python 3.12 so to make it work with the script a modification to the `create-venv.sh` script would look something like this:

```sh
#!/usr/bin/env bash

function create_venv () {
  if [[ ! -z "${VIRTUAL_ENV:-}" ]]; then
    echo "Virtualenv is already active! Run 'deactivate' to deactivate the virtualenv."
    return 0
  fi

## finding the current python3 version
  version=`python3 --version | sed 's/Python //'` && pythoncmd=python${version%.*}

  echo "Creating virtualenv"
  $pythoncmd -m pip install install --user virtualenv==20.24.6

  NILLION_VENV=".venv"
  mkdir -p "$NILLION_VENV"
  $pythoncmd -m virtualenv -p $pythoncmd "$NILLION_VENV"
  source "$NILLION_VENV/bin/activate"
  $pythoncmd -m pip install -r requirements.txt

  echo "Virtualenv: $NILLION_VENV"
  echo "Check the $NILLION_VENV/lib/$pythoncmd/site-packages folder to make sure you have py_nillion_client and nada_dsl packages"
}

create_venv
```

**Comment by cmacrae** [comment]
thanks for taking the time to report these issues @eum602 ! I've raised NillionNetwork/nillion-docs#25 & NillionNetwork/scaffold-nillion#13 to address these
